### PR TITLE
Refactor workflow job

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Review by trantienduchn — APPROVED** [review]

File: workflow-job [hunk 1]
```diff
@@ -325,379 +364,378 @@ def provisionDataForIntegrationTest(dockerContainer, jamesCliCmd) {
 }
 
 def containerIp(dockerContainer) {
-    return "`docker inspect -f '{{(index .NetworkSettings.IPAddress)}}' ${dockerContainer}`" 
+    return "`docker inspect -f '{{(index .NetworkSettings.IPAddress)}}' ${dockerContainer}`"
 }
 
+// Define main variables
+echo "Defining variables"
```
Comment (rouazana) [inline workflow-job]: debug?

File: workflow-job [hunk 2]
```diff
@@ -325,379 +364,378 @@ def provisionDataForIntegrationTest(dockerContainer, jamesCliCmd) {
...
+if (publishToDocker()) {
+    publishing = commitStatusFactory.from("Publishing");
+    statuses.addPendingStatus(publishing);
+}
+packagingTrigger;
+if (packagageOnGitlab()) {
+    packagingTrigger = commitStatusFactory.from("Triggering packaging");
+    statuses.addPendingStatus(packagingTrigger);
+}
+echo "Variables defined"
```
Comment (rouazana) [inline workflow-job]: debug?

**Review by rouazana — APPROVED** [review]

File: workflow-job [hunk 3]
```diff
@@ -11,11 +11,11 @@ import org.kohsuke.github.GHRepository
 // $sha1 parameter: specifies the actual commit
 // $originalBranch: the branch used for merge (optional)
 
-def hostname = 'james.linagora.com'
-def maxRetries = 120
-def author = extractUsername()
-def authorParentURL = "https://github.com/${author}/james-project.git"
-def repoName = extractRepoName()
+hostname = 'james.linagora.com'
```
Comment (mbaechler) [inline workflow-job:14]: I don't understand this change, reading http://groovy-lang.org/semantics.html#_variable_definition:
```
def is a replacement for a type name. In variable definitions it is used to indicate that you don’t care about the type. In variable definitions it is mandatory to either provide a type name explicitly or to use "def" in replacement. This is needed to make variable definitions detectable for the Groovy parser.
```
So it should be either `def` or `String` but I don't understand the syntax you chose

File: workflow-job [hunk 4]
```diff
@@ -139,9 +140,23 @@ public class Images implements Serializable {
         jamesSpring = 'james-server-spring-' + buildId;
         gitPublish = 'git/publish-' + buildId;
     }
+
+    def clean = { node ->
```
Comment (mbaechler) [inline workflow-job]: s/clean/deleteAll ?

File: workflow-job (hunk 3 above)
Comment (aduprat) [reply workflow-job:14]: It's a way for defining global variables in groovy, here is a long explanation:
https://www.michaeloland.com/2018/02/groovy.basic.binding.scope.html

**Review by rouazana — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**